File: Data Structures & Algorithms/is-palindrome/submission-7.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int start = 0;
        int end = s.length() - 1;

        while (end > start) {
            if (!iswalnum(s[start])) {
                start++;
                continue;
            }
            else if (!iswalnum(s[end])) {
                end--;
                continue;
            }
            char first = (char)tolower(s[start]);
            char last = (char)tolower(s[end]);

            if (first != last) {
                return false;
            }
            else {
                start++;
                end--;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-7.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res = ""
        for string in strs:
            res += str(len(string))
            res += "#"
            res += string
        return res
        
    def decode(self, s: str) -> List[str]:
        print(s)
        res = []
        start = 0
        end = start

        while (start < len(s) - 1):
            while (s[end] != "#" and end < len(s) - 1):
                end += 1
            length = s[start:end]
            print(length)
            beginning = end + 1
            last = beginning + int(length)
            string = s[beginning:last]
            res.append(string)
            start = last
            end = start
        return res